polaris/hub/client.py

```diff
@@ -153,22 +154,28 @@ def _load_from_signed_url(self, url: URLTypes, load_fn: Callable, headers: Heade
         content = BytesIO(response.content)
         return load_fn(content)
 
-    def _base_request_to_hub(self, url: str, method: str, **kwargs):
+    def _base_request_to_hub(self, url: str, method: str, show_informative_error: bool = False, **kwargs):
```

Comment:
I added this early on when I was under the impression that only some client methods would be impacted by this change. Not the case though, good catch!

This does make me think though, the list operations will remain silently impacted by this data discrepancy issue. Since they do not throw 403s and they instead just don't return names of artifacts the user doesn't have access to, we can't intervene (as is) and suggest a re-login. 

While we made other changes to mitigate this (e.g. TTL), it is something we may want to consider in the near future.

Reply:
Good point. I don't think it will be too big an issue (simply trying again a few minutes later would solve it), but it could indeed be confusing. 

More generally, with these PRs I've been thinking a bit more about how we can streamline error handling. I think there is a lot to gain here, e.g. also by parsing the Zod error message the Hub returns in case of a 409. At the summer school, I felt that a lot of questions arose due to people not understanding the stack trace.